pkg/controller/accountclaim/accountclaim_controller.go

```diff
@@ -209,7 +209,17 @@ func (r *ReconcileAccountClaim) Reconcile(request reconcile.Request) (reconcile.
 	if accountClaim.Spec.AccountLink == "" {
 		setAccountLinkOnAccountClaim(reqLogger, unclaimedAccount, accountClaim)
 		return reconcile.Result{}, r.specUpdate(reqLogger, accountClaim)
+	}
 
+	if !accountClaim.Spec.ManualSTSMode {
```

Comment:
This code block seems a duplicate from the code block below, would it make sense to move them to a func if they're doing the same thing?

Reply:
Updated